crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -68,6 +74,14 @@ impl<'a> Compressor<'a> {
                 UnaryOperator::Typeof => {
                     self.try_fold_typeof(unary_expr.span, &unary_expr.argument)
                 }
+                UnaryOperator::UnaryPlus
+                | UnaryOperator::UnaryNegation
+                | UnaryOperator::LogicalNot
+                | UnaryOperator::BitwiseNot
+                    if !has_side_effects =>
```

Comment:
better to inline this? `if !expr.may_have_side_effects()`?

Reply:
I encountered this error, so I moved it to the top of `fold_expression` 😭:
```
cannot borrow `*expr` as immutable because it is also borrowed as mutable
immutable borrow occurs hererustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B2%5D?2#file%3A%2F%2F%2FUsers%2Fwenzhe%2Ffork%2Foxc%2Fcrates%2Foxc_minifier%2Fsrc%2Fcompressor%2Ffold.rs)
fold.rs(72, 41): mutable borrow occurs here
fold.rs(82, 50): mutable borrow later used here
```

Reply:
Ohh ... you need to implement the trait on UnaryExpression, and call `unary_expr. may_have_side_effects()`